### PR TITLE
docs: use lazy hoverxref

### PR DESCRIPTION
**Pull request by bast0006**
## Summary

This PR busts the CSS/JS caches because despite RTD's documentation stating they "bust the cache" every time a new version is submitted, there have been reports of users with the unfortunate experience of having old JS running with newer html.

This PR also replaces the live version of sphinx-hoverxref with a forked version that has been altered to utilize lazy-loaded hoverxref hooks instead of on-page-load, which should dramatically speed up page load times.

## Checklist

- [x] If code changes were made, then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [ ] I have formatted the code properly by running `task lint`
    - [ ] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: setup.py
Comment (onerandomusername) [inline setup.py:46]: The problem is we won't be able to merge this pr until this is on pypi. Pypi requires all dependencies of projects on pypi also be on pypi.

**Review by onerandomusername — CHANGES_REQUESTED** [review]
The change looks good but we have to wait for a pypi release of sphinx-hoverxref.

**Comment by onerandomusername** [comment]
@bast0006 any status on when the pr to sphinx-hoverxref will be made?

**Comment by onerandomusername** [comment]
update: the pr has been made: https://github.com/readthedocs/sphinx-hoverxref/pull/227

File: docs/_templates/layout.html
Comment (onerandomusername) [inline docs/_templates/layout.html:15]: its been several months, the cache bust is no longer necessary.

Comment (onerandomusername) [reply docs/_templates/layout.html:15]: edit: nvm

**Comment by onerandomusername** [comment]
Due to the cache bust and other general speedups here, I'd *like* to merge this after #722 unless there is an extended period after this pr is ready and #722 is still not ready.

**Comment by shiftinv** [comment]
> Due to the cache bust

that only affects select files and shouldn't be relevant to #722, I think.
It might be worth consideration to actually do that for all js files, may be difficult to do with sphinx though.

**Comment by onerandomusername** [comment]
This PR as it is currently breaks the dark theme toggle.

**Comment by bast0006** [comment]
It's just the cache bust doing it (sigh). Pushing a fix shortly.

**Review by onerandomusername — APPROVED** [review]
lgtm

File: docs/_static/custom.js
Comment (shiftinv) [inline docs/_static/custom.js:137]:
```suggestion
```

**Review by shiftinv — APPROVED** [review]
other than this tiny change, lgtm. Thanks for adding this to hoverxref upstream!

**Pull request merged**
